symfony/plugins/orangehrmAdminPlugin/modules/admin/actions/deleteEmploymentStatusAction.class.php

```diff
@@ -19,33 +19,35 @@
  */
 
 class deleteEmploymentStatusAction extends sfAction {
-	
-	public function getEmploymentStatusService() {
-		if (is_null($this->empStatusService)) {
-			$this->empStatusService = new EmploymentStatusService();
-			$this->empStatusService->setEmploymentStatusDao(new EmploymentStatusDao());
-		}
-		return $this->empStatusService;
-	}
-	
-	public function execute($request) {
-                $form = new DefaultListForm();
-                $form->bind($request->getParameter($form->getName()));
-		$toBeDeletedStausIds = $request->getParameter('chkSelectRow');
-
-		if (!empty($toBeDeletedStausIds)) {
-
-			foreach ($toBeDeletedStausIds as $toBeDeletedStausId) {
-                            if ($form->isValid()) {
-				$status = $this->getEmploymentStatusService()->getEmploymentStatusById($toBeDeletedStausId);
-				$status->delete();
-                                $this->getUser()->setFlash('success', __(TopLevelMessages::DELETE_SUCCESS));
-                            }
-			}			
-		}
-
-		$this->redirect('admin/employmentStatus');
-	}
+
+    public function getEmploymentStatusService() {
+        if (is_null($this->empStatusService)) {
+            $this->empStatusService = new EmploymentStatusService();
+            $this->empStatusService->setEmploymentStatusDao(new EmploymentStatusDao());
+        }
+        return $this->empStatusService;
+    }
+
+    public function execute($request) {
+        $form = new DefaultListForm();
+        $form->bind($request->getParameter($form->getName()));
+        $toBeDeletedStausIds = $request->getParameter('chkSelectRow');
+
+        if (!empty($toBeDeletedStausIds)) {
+
+            foreach ($toBeDeletedStausIds as $toBeDeletedStausId) {
+                if ($form->isValid()) {
+                    $status = $this->getEmploymentStatusService()->getEmploymentStatusById($toBeDeletedStausId);
+                    if ($status instanceof EmploymentStatus) {
+                        $status->delete();
+                    }
+                    $this->getUser()->setFlash('success', __(TopLevelMessages::DELETE_SUCCESS));
```

Comment:
Flash in foreach loop?
flash looks like it will be set 20 times if 20 Ids are passed?